kernel/tests/read.rs

```diff
@@ -503,23 +516,111 @@ fn column_ordering_and_projection() -> Result<(), Box<dyn std::error::Error>> {
     Ok(())
 }
 
-#[test]
-fn predicate_on_number() -> Result<(), Box<dyn std::error::Error>> {
-    let expected = vec![
+// get the basic_partitioned table for a set of expected numbers
+fn table_for_numbers(nums: Vec<u32>) -> Vec<String> {
+    let mut res: Vec<String> = vec![
         "+---------+--------+",
         "| a_float | number |",
         "+---------+--------+",
-        "| 1.1     | 1      |",
-        "| 2.2     | 2      |",
-        "| 3.3     | 3      |",
-        "+---------+--------+",
+    ]
+    .into_iter()
+    .map(String::from)
+    .collect();
+    for num in nums.iter() {
+        res.push(format!("| {num}.{num}     | {num}      |"));
+    }
+    res.push("+---------+--------+".to_string());
+    res
+}
+
+#[test]
+fn predicate_on_number() -> Result<(), Box<dyn std::error::Error>> {
+    let cases = vec![
+        (
+            Expression::Column("number".to_string()).lt(Expression::Literal(Scalar::Long(4))),
+            table_for_numbers(vec![1, 2, 3]),
+        ),
+        (
+            Expression::Column("number".to_string()).le(Expression::Literal(Scalar::Long(4))),
+            table_for_numbers(vec![1, 2, 3, 4]),
+        ),
+        (
+            Expression::Column("number".to_string()).gt(Expression::Literal(Scalar::Long(4))),
+            table_for_numbers(vec![5, 6]),
+        ),
+        (
+            Expression::Column("number".to_string()).ge(Expression::Literal(Scalar::Long(4))),
+            table_for_numbers(vec![4, 5, 6]),
+        ),
+        (
+            Expression::Column("number".to_string()).eq(Expression::Literal(Scalar::Long(4))),
+            table_for_numbers(vec![4]),
+        ),
+        (
+            Expression::Column("number".to_string()).ne(Expression::Literal(Scalar::Long(4))),
+            table_for_numbers(vec![1, 2, 3, 5, 6]),
+        ),
     ];
-    read_table_data(
-        "./tests/data/basic_partitioned",
-        Some(&["a_float", "number"]),
-        Some(Expression::Column("number".to_string()).lt(Expression::Literal(Scalar::Long(4)))),
-        expected,
-    )?;
+
+    for (expr, expected) in cases.into_iter() {
+        read_table_data(
+            "./tests/data/basic_partitioned",
+            Some(&["a_float", "number"]),
+            Some(expr),
+            expected,
+        )?;
+    }
+    Ok(())
+}
+
+#[test]
+fn predicate_on_number_not() -> Result<(), Box<dyn std::error::Error>> {
+    let cases = vec![
+        (
+            Expression::Column("number".to_string())
+                .lt(Expression::Literal(Scalar::Long(4)))
+                .not(),
```

Comment:
I find this form of `not` rather non-intuitive. Perhaps better to just do functional form instead?
```suggestion
            not(lt(column("number"), literal(4i64)),
```